extension_registry_gui.py

```
@@ -0,0 +1,1368 @@
+        if search_text:
+            extensions = [ext for ext in extensions if 
+                         search_text in ext['extension'].lower() or 
+                         (ext['description'] and search_text in ext['description'].lower())]
+        
+        # Filter based on category
+        if category_filter != "All Categories":
+            extensions = [ext for ext in extensions if ext['category_name'] == category_filter]
+        
+        self.populate_extensions_table(extensions)
+    
+    def add_extension(self):
```

Ruff (0.12.2)

Redefinition of unused `add_extension` from line 635

(F811)